### PR TITLE
Fixes #717 - Don't allow saving of empty Group name

### LINKED ISSUE #717: Bookmark Groups should be validated as having non-zero length titles before being created
### Expected behavior

A bookmark group without a name cannot be created.

### Actual behavior

It can be created, which causes other problems later on.

### Steps to reproduce the behavior

Create a bookmark group, leave the name blank.

### App Version/Hardware

2.6.0

### PR DESCRIPTION
**Pull request by chadsy**
* Elected to fix this by enabling/disabling the Save button based on text field length
* Also collapsed redundant UIAlertController code into one method for add/edit

### CODE REVIEW

**Comment by chadsy** [comment]
@aaronbrethorst if you're not a fan of this solution, i can duplicate the fix for the bookmark names (show an error).

**Comment by aaronbrethorst** [comment]
I think you're going to have to implement this on the model and check that instead. There's a whole other experience that's accessible via Stop -> Add Bookmark -> Select Route -> Group -> Add. Both should have the same validation for their models. lmk if you have any questions.

**Comment by chadsy** [comment]
Thanks for pointing out the additional case. Added some discussion to the bug, will proceed as described there and update this PR.

**Comment by aaronbrethorst** [comment]
@chadsy don't forget you need to git fetch upstream, then rebase against upstream/develop, and then push -f

File: ui/bookmark_groups/OBABookmarkGroupsViewController.m [hunk 1]
```diff
@@ -58,24 +58,53 @@ - (void)close {
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
-- (void)addGroup {
-    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Add Bookmark Group", @"") message:nil preferredStyle:UIAlertControllerStyleAlert];
+static UIAlertAction *saveButton;
```
Comment (aaronbrethorst) [inline ui/bookmark_groups/OBABookmarkGroupsViewController.m]: Why not make this a property on the object? If you want to keep this as is, please set to nil initially.

File: ui/bookmark_groups/OBABookmarkGroupsViewController.m [hunk 2]
```diff
@@ -58,24 +58,53 @@ - (void)close {
...
+
+- (void)textFieldDidChange :(UITextField *)textField {
```
Comment (aaronbrethorst) [inline ui/bookmark_groups/OBABookmarkGroupsViewController.m]: Wonky spacing around param

**Comment by aaronbrethorst** [comment]
looks great, thank you! 🎆

**Pull request merged**